providers/fips/self_test_kats.c

```diff
@@ -689,9 +696,16 @@ static int self_test_kas(OSSL_SELF_TEST *st, OSSL_LIB_CTX *libctx)
 static int self_test_signatures(OSSL_SELF_TEST *st, OSSL_LIB_CTX *libctx)
 {
     int i, ret = 1;
-
-    for (i = 0; i < (int)OSSL_NELEM(st_kat_sign_tests); ++i) {
-        if (!self_test_sign(&st_kat_sign_tests[i], st, libctx))
+    const ST_KAT_SIGN *t;
+
+    for (i = 0; ret && i < (int)OSSL_NELEM(st_kat_sign_tests); ++i) {
+        t = st_kat_sign_tests + i;
+        if (!set_kat_drbg(libctx, t->entropy, t->entropy_len,
```

Comment:
doing these as OSSL_PARAM might be nicer..

Reply:
That's what happened inside that function.  Why add additional mess here?

Reply:
I mean not using DRBG, but using new params setters that would be added to core_names.h and implemented in a pkey. (Up to you what method you use),

Reply:
That ends up pretty ugly.  The RNGs are accessed rather deep down and intercepting things too high up isn't testing the key gen.

I'm thinking about setting up a separate libctx for these test cases, although I'm not sure how workable it is at this point.

I can verify that the DRBGs are restored correctly outside of the tests if extra assurance is required.

Reply:
> I can verify that the DRBGs are restored correctly outside of the tests if extra assurance is required.

Please do.

Reply:
Done, FWIW.